File: 5 Semestre/Fundamentos de Computação Concorrente, Paralela e Distribuída/Atividades/Atividade Segurança/pratica-jwt/tools/generate-token.py
#!/usr/bin/env python3
"""
Script para gerar tokens JWT
Uso: python generate-token.py [tempo_expiracao]
Exemplo: python generate-token.py 60
"""

import jwt
import sys
from datetime import datetime, timedelta

# Chave compartilhada (deve ser a mesma do gateway)
SECRET_KEY = 'MINHA_CHAVE_DE_ASSINATURA'

# Tempo de expiração (em segundos)
expires_in_seconds = int(sys.argv[1]) if len(sys.argv) > 1 else 30

# Payload do token
payload = {
    'service': 'service-a',
    'timestamp': datetime.utcnow().isoformat(),
    'purpose': 'inter-service-communication',
    'exp': datetime.utcnow() + timedelta(seconds=expires_in_seconds)
}

# Gerar token
token = jwt.encode(payload, SECRET_KEY, algorithm='HS256')

print('\n🔑 JWT Token Gerado com Sucesso!\n')
print('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━')
print('Token:')
print(token)
print('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n')

print('📋 Informações:')
print(f'   Expira em: {expires_in_seconds} segundos')
print(f'   Algoritmo: HS256')
print(f'   Payload: {payload}')

print('\n💡 Como usar:\n')
print(f'   # Com curl:')
print(f'   curl -H "Authorization: Bearer {token}" http://localhost:8000/minha-rota\n')

print(f'   # Com PowerShell:')
print(f'   $headers = @{{ "Authorization" = "Bearer {token}" }}')
print(f'   Invoke-RestMethod -Uri "http://localhost:8000/minha-rota" -Headers $headers\n')

# Decodificar token para mostrar
decoded = jwt.decode(token, SECRET_KEY, algorithms=['HS256'])
print('🔍 Token decodificado:')
print(f'   Payload: {decoded}')
print(f'   Expira em: {datetime.fromtimestamp(decoded["exp"]).isoformat()}')

File: 5 Semestre/Fundamentos de Computação Concorrente, Paralela e Distribuída/Atividades/Atividade Segurança/pratica-jwt/tools/generate-token.js
const jwt = require('jsonwebtoken');

// Chave compartilhada (deve ser a mesma do gateway)
const SECRET_KEY = 'MINHA_CHAVE_DE_ASSINATURA';

// Configurações
const expiresIn = process.argv[2] || '30s'; // Pode passar como argumento: node generate-token.js 60s

// Payload do token
const payload = {
  service: 'service-a',
  timestamp: new Date().toISOString(),
  purpose: 'inter-service-communication'
};

// Gerar token
const token = jwt.sign(payload, SECRET_KEY, { 
  expiresIn: expiresIn,
  algorithm: 'HS256'
});

console.log('\n🔑 JWT Token Gerado com Sucesso!\n');
console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━');
console.log('Token:');
console.log(token);
console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n');

console.log('📋 Informações:');
console.log(`   Expira em: ${expiresIn}`);
console.log(`   Algoritmo: HS256`);
console.log(`   Payload:`, payload);

console.log('\n💡 Como usar:\n');
console.log('   # Com curl:');
console.log(`   curl -H "Authorization: Bearer ${token}" http://localhost:8000/minha-rota\n`);

console.log('   # Com PowerShell:');
console.log(`   $headers = @{ "Authorization" = "Bearer ${token}" }`);
console.log(`   Invoke-RestMethod -Uri "http://localhost:8000/minha-rota" -Headers $headers\n`);

// Decodificar token para mostrar
const decoded = jwt.decode(token, { complete: true });
console.log('🔍 Token decodificado:');
console.log('   Header:', decoded.header);
console.log('   Payload:', decoded.payload);
console.log('   Expira em:', new Date(decoded.payload.exp * 1000).toISOString());
